src/library/scala/collection/Map.scala

```diff
@@ -199,10 +200,11 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
     override def isEmpty: Boolean = MapOps.this.isEmpty
   }
 
-  /** Collects all keys of this map in an iterable collection.
-    *
-    *  @return the keys of this map as an iterable.
-    */
+  /** An [[Iterable]] collection of the keys contained by this map.
+   *
+   *  @return  an [[Iterable]] collection of the keys contained by this map.
+   */
+  @deprecatedOverriding("A set view of keys, i.e., keySet; for an ordered traversal, use keysIterator or keys.iterator", since="2.13.13")
   def keys: Iterable[K] = keySet
```

Comment:
Should `keys` be deprecated? Why does it exist alongside `keySet` and `keysIterator`?

Reply:
I tried just deprecation first and it was bad FSR. I think a couple of collections wanted to offer an alternative where keySet did not feel natural or showed mismatched impedance.

Reply:
I noticed that I commented on the PR that keys is popular (because it is a short name, presumably). I propose we introduce an alias `keyS` to emphasize that it is just a short form of the name `keySet`.